### ISSUE TITLE
Favor TTS over Recorded voices

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#17683]
Carry over of e.g  https://github.com/osmandapp/OsmAnd/issues/17645#issuecomment-1640250985: We have a number of issues where users complain about weaknesses of recorded voices who should likely have used the more powerful TTS in the first place. It seems we should revisit all voice selection/proposal screens and documentation to check if we really do enough to make all users who can use TTS. Reserving recorded voices for rarer special or fallback cases only.

- [ ] Installation default and/or the dialog upon first navigation use. Do we by default suggest a TTS voice? Do we explain that recorded voices are a poor(er) fallback only? Do we perhaps need a check if the user device setup is TTS capable for the user's system language?
- [ ] On re-visits of screen Navigation settings > Voice prompts > Language, it seems we display the last visited tab, but should always use the TTS tab first?
- [ ] Should we maybe  change our overall wording to "Standard voices (TTS)" and "Fallback voices (Pre-recorded)" or something similar?

**Comment by vshcherb** [comment]
Not sure how massive the issue is. For some languages it's definitely easier to delete recorded packages and hide them completely allow only to download on the website

**Comment by sonora** [comment]
Yes, or perhaps simply some string work like the third bullet above.

**Comment by vshcherb** [comment]
Well string is still responsibility ;-) I personally think that we should check how good TTS languages on phones without Google and simply delete, leave only for the server tests and for languages that doesn't exist yet.

**Comment by GabrielSebastianMoise** [comment]
I use Voci Vocalizer TTS for Romanian language, and works so great. Google
TTS has a weird pronunciation.

mar., 25 iul. 2023, 23:33 vshcherb ***@***.***> a scris:

**Comment by sonora** [comment]
Looking at this list, it seems there are hardly any languages not TTS supported at all in Android: https://accessibleandroid.com/list-of-languages-with-available-tts-engines-on-android/.

There seem 3 basic constellations:
- **(A) User device already TTT-supports the language in question.**
  - User may at most have to download a corresponding TTS language pack yet. (Perhaps we can assist via intent ACTION_INSTALL_TTS_DATA.)
  - [ ] _This use case I am primarily concerned about. Where we don't prevent it, users may unknowingly and unnecessearily use OsmAnd without TTS._

- **(B) User device would support the language in question, via the google, the manufacturer's or a third patty TTS engine**, see e.g. list above.
  - But user may still need to activate, or even identify, download and install the appropriate TTS engine and a corresponding language pack.
  - [ ] _(B) is a challenge for many users, any OsmAnd or documentation support may prevent users from falling back to a less functional OsmAnd config using a recorded voice._

- **(C) There is no TTS support for the language in question.**
  - [ ] _I am not sure this situation still exists and how to identify it. It seems the only use case where we should readily offer a recorded voice at all in OsmAnd?_


Some intro reading is here  https://android-developers.googleblog.com/2009/09/introduction-to-text-to-speech-in.html?m=1, some of the mentioned return codes we already use in our Test voice prompt button 11.1.
